src/utils/schema.ts

```diff
@@ -28,3 +32,47 @@ export function parseSchemaResult(schema: z.ZodTypeAny, v: unknown) {
   }
   return schemaResult;
 }
+
+export async function validateAnt({
+  processId,
+  ao = connect(),
+}: {
+  processId: string;
+  ao: AoClient;
+}): Promise<Record<AoANTHandler, { valid: boolean; error?: string }>> {
+  const antHandlers = await getAntHandlers({ processId, ao });
+  const defaultValidations = Object.fromEntries(
+    AntHandlerNames.map((handler) => [
+      handler,
+      async (_: { processId: string; ao: AoClient }) => {
+        if (antHandlers.includes(handler)) return true;
+        throw new Error(`Handler ${handler} not found`);
+      },
+    ]),
+  ) as Record<
+    AoANTHandler,
+    (p: { processId: string; ao: AoClient }) => Promise<boolean>
+  >;
+  const handlerValidationConfig: Record<
+    AoANTHandler,
+    (p: { processId: string; ao: AoClient }) => Promise<boolean>
+  > = {
+    ...defaultValidations,
+    // Add custom validations here for more complex checks using dryrun
+  };
+
+  const results: Record<AoANTHandler, { valid: boolean; error?: string }> =
+    {} as Record<AoANTHandler, { valid: boolean; error?: string }>;
```

Comment:
in regards to the comment about - we can declare that the function must return an object like this


```typescript
export const validateANT({ processId, ao, validations }: { processId: string, ao: connect, validations: () => Record<AoANTHandler, valid: boolean: error: string | undefined }>;
```